### PR TITLE
Do not focus input after other interactions

### PR DESCRIPTION
**Pull request by rezkiy37**
### Details

The PR fixes the Composer focusing and keyboard behavior after interactions with modals.

### Fixed Issues

$ https://github.com/Expensify/App/issues/47082
PROPOSAL: https://github.com/Expensify/App/issues/47082#issuecomment-2326318309

### Tests

1. Open any chat.
2. Tap the "**+**" button. 
3. Tap on the backdrop to close the modal.
4. Verify that Composer has not been focused and the keyboard is not opened.
5. Tap the "**+**" button > **Add attachment**.
6. Tap on the backdrop to close the modal.
7. Verify that Composer has not been focused and the keyboard is not opened.

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as "**Tests**".

### QA Steps

Same as "**Tests**".

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/cafc72be-6e90-44c2-a321-47d0036b07dd

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/cfa7676f-1916-4e3b-a557-14356f0f15ba

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/583d3c3e-dcbf-4131-b375-1ae77a6a9d23

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/1616458e-d74c-4ec5-9c4c-4c13eb48ac27

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/4255085d-b344-4c16-8556-e466910f1b99

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/fe9cb767-32f9-4fae-b641-45f8bd0ad626

</details>

### CODE REVIEW

File: src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx [hunk 1]
```diff
@@ -559,12 +539,6 @@ function ReportActionCompose({
                             {DeviceCapabilities.canUseTouchScreen() && isMediumScreenWidth ? null : (
                                 <EmojiPickerButton
                                     isDisabled={isBlockedFromConcierge || disabled}
-                                    onModalHide={(isNavigating) => {
-                                        if (isNavigating) {
-                                            return;
-                                        }
-                                        focus();
-                                    }}
```
Comment (fabioh8010) [inline src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx:567]: Just curious, does the same problem happen with the emoji picker?

Comment (rezkiy37) [reply src/pages/home/report/ReportActionCompose/ReportActionCompose.tsx:567]: Yes, the same one because Composer gets focused after that.

**Review by fabioh8010 — APPROVED** [review]
LGTM, but I think we should remove `[No QA]` label as this PR should have tests

**Comment by rezkiy37** [comment]
@eh2077, please review the PR 🙂

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx [hunk 2]
```diff
@@ -653,9 +653,7 @@ function ComposerWithSuggestions(
 
         if (editFocused) {
             InputFocus.inputFocusChange(false);
-            return;
         }
-        focus(true);
```
Comment (eh2077) [inline src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:658]: By removing this, we also won't focus back to the composer when navigating back from RHP. I just feel that the impact scope might be wider than we expect.


https://github.com/user-attachments/assets/822bbff3-d020-44af-bd19-93bf6ae7ce97


https://github.com/user-attachments/assets/031f7117-0efb-4b5d-affb-710a0b7cf6f8

**Review by eh2077 — COMMENTED** [review]
I'm not sure if we should only avoid keyboard focusing for attachment picker modals. 

And I feel that we have been so keen on the auto-focus feature in the past.

File: src/components/EmojiPicker/EmojiPickerButton.tsx [hunk 3]
```diff
@@ -24,12 +24,12 @@ type EmojiPickerButtonProps = {
...
     shiftVertical?: number;
 
-    onModalHide: EmojiPickerAction.OnModalHideValue;
+    onModalHide?: EmojiPickerAction.OnModalHideValue;
 
     onEmojiSelected: EmojiPickerAction.OnEmojiSelected;
 };
 
-function EmojiPickerButton({isDisabled = false, id = '', emojiPickerID = '', shiftVertical = 0, onModalHide, onEmojiSelected}: EmojiPickerButtonProps) {
+function EmojiPickerButton({isDisabled = false, id = '', emojiPickerID = '', shiftVertical = 0, onModalHide = () => {}, onEmojiSelected}: EmojiPickerButtonProps) {
```
Comment (eh2077) [inline src/components/EmojiPicker/EmojiPickerButton.tsx:32]: After selecting an emoji, we won't focus back the composer. TBH, I'm feeling a bit disquiet about changing this. 

https://github.com/user-attachments/assets/c1e57a76-893f-44d2-bb09-984c2c45d411

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx (hunk 2 above)
Comment (rezkiy37) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:658]: I see. To clarify: we have the problem on the Android app only. So we can apply these changes to Android and keep the other platforms as they are. WDYT @eh2077?

cc @mountiny

File: src/components/EmojiPicker/EmojiPickerButton.tsx (hunk 3 above)
Comment (rezkiy37) [reply src/components/EmojiPicker/EmojiPickerButton.tsx:32]: It is also a part of the scope of fixing. From my point of view, it is not a problem and it is good as well. But let's decide https://github.com/Expensify/App/pull/48415#discussion_r1751702038 first.

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx (hunk 2 above)
Comment (eh2077) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:658]: Hmm, as a user, I'd be happy if keyboard is not focused (no matter the keyboard is focused before or not) when navigating back from **_attachment picker modals_**, no matter after uploading an attachment or just cancel to pick an attachment. 

For other cases, like picking up an emoji or back from RHP, I think the current behaviour to focus input makes senses to me. I'd like to share following points to support my stance:

- Back the issue itself, the delay is more noticeable only after interactions with system, like open camera or media manager
- Auto-focusing the keyboard makes the user more engaged in chatting, like after picking up an emoji
- IIRC, we have done quite a lot of polishing to this feature in the past.

@mountiny We'd also like to hear your thoughts on this.

Comment (mountiny) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:658]: I _think_ I agree with @eh2077 but I would like to give this more exposure, can you please summaarize the problem now and post what you suggest how we should handle it in Slack? I think we should get design team to agree on this

Comment (rezkiy37) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:658]: I totally agree with aligning it with the design team. I will create a post in the open-source channel.

Comment (rezkiy37) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:658]: https://expensify.slack.com/archives/C01GTK53T8Q/p1726064308789469

Comment (rezkiy37) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:658]: @eh2077, can we move forward with this PR after the discussion in Slack where the guys said that auto-focus feels unnecessary?

Comment (mountiny) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:677]: @rezkiy37 I think we need to fix the case @eh2077 raised in this comment. in Web, we want to make sure the composer i focused when you go back from the RHP as that is a common flow and I think lots of users would be unhappy if we stop focusing to the composer

Comment (rezkiy37) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx:677]: Okay, I will do it. Thank you!

**Comment by eh2077** [comment]
I'd like to pull the design team @Expensify/design  here, so we can hear feedback from them on new changes

File: src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx (hunk 2 above)
Comment (rezkiy37) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx]: Done - 3c4b392.

https://github.com/user-attachments/assets/87da87b7-3e4b-4165-88f9-5aa79fe615f6

Comment (rezkiy37) [reply src/pages/home/report/ReportActionCompose/ComposerWithSuggestions/ComposerWithSuggestions.tsx]: https://github.com/user-attachments/assets/1278fdd9-445e-47e3-8efa-2d7755e092ed

**Comment by eh2077** [comment]
> @eh2077 @rezkiy37 to make this easier for the design team, can you provide before and after videos of the places that are affected?

Sure, I'll upload videos to compare with staging.

**Comment by mountiny** [comment]
@rezkiy37 can you make sure this is up to date with main? maybe your fork main is out of date, not sure why the adhoc build is failing

**Comment by rezkiy37** [comment]
@mountiny, I've synced it up.

**Comment by mountiny** [comment]
@rezkiy37 @eh2077 can you please try to provide the comparison videos too? thanks!

**Comment by dubielzyk-expensify** [comment]
I just tried the build and it looks good to me. I was worried about the case where you have the compose bar focused and keyboard up then pulled up a modal and closed it, but I think dismissing the focus there also makes sense.

Keen to hear what the other @Expensify/design thinks

**Comment by rezkiy37** [comment]
> @rezkiy37 @eh2077 can you please try to provide the comparison videos too? thanks!

<details><summary>Before</summary>

https://github.com/user-attachments/assets/59a8f42e-043a-4b27-add1-e3e3ce8907a3


https://github.com/user-attachments/assets/ded5e28e-922d-4105-a938-ed6b21994bb7


https://github.com/user-attachments/assets/ff8a1379-ec0d-4729-bd3e-ba60a18fa8ad


</details> 

<details><summary>After</summary>

https://github.com/user-attachments/assets/426db73a-e5a2-47de-80d3-e4b15501f6e3


https://github.com/user-attachments/assets/00707330-744a-4b69-8004-5a7ae4d24c51


https://github.com/user-attachments/assets/6065c232-97bf-434f-acc4-a5ec8334963c


</details> 


cc @mountiny @dubielzyk-expensify @Expensify/design(ers)

**Comment by shawnborton** [comment]
Hmm if I'm being honest, I'm not quite sure the differences between before and after 🤣 do you mind spelling that out for me? Thanks!

**Comment by shawnborton** [comment]
For iOS, I can see the improvement where we don't get the jump of having the keyboard whip back open. I think that's a nice improvement.

**Comment by rezkiy37** [comment]
> Hmm if I'm being honest, I'm not quite sure the differences between before and after on web... 🤣 do you mind spelling that out for me? Thanks!

I agree that the prev. video is not a good example. I've replaced it with this one: 

https://github.com/user-attachments/assets/f44386c5-b7ab-4b66-8ee8-d1875a52af1f

**Comment by dannymcclain** [comment]
Yeah this feels like an improvement on mobile at least. Definitely feels a lot less jumpy. It will be nice to experience it and see what it feels like for real.

**Comment by eh2077** [comment]
On web Chrome, the input will be focused again when return from attachment picker.
- Staging: Yes
- Dev: No


https://github.com/user-attachments/assets/5a40750e-f466-4908-90f7-b91a542e47f8



---

On web Chrome, the input will be focused again when return from 😄emoji😄  picker
- Staging: Yes
- Dev: No



https://github.com/user-attachments/assets/6ea56e09-d293-4238-8eac-027c51327013


@Expensify/design Do we want to change the two behaviours?

**Comment by dannymcclain** [comment]
I'm not sure I'm totally following (sorry, the differences are pretty subtle on desktop!) but generally I'm not as concerned about desktop behavior because since there's no keyboard that pops up, the UI doesn't really shift at all when the composer is focused. @shawnborton any strong feelings here?

**Comment by shawnborton** [comment]
I feel like if you are typing, then you pick an emoji, it makes sense that you would probably continue typing in that same input... so the auto focus there makes sense to me. I feel a little less strongly about the attachment picker since you technically clicked outside of the text input portion of the composer.

**Comment by rezkiy37** [comment]
Alright, I will revert a change that autofocus after emojis.

**Comment by rezkiy37** [comment]
I've reverted the change to emoji modal interactions. It focuses `Composer` once the modal closes.

<details><summary>Details</summary>

https://github.com/user-attachments/assets/8b1be405-e4a3-43c1-a3ff-ae46198b8758

https://github.com/user-attachments/assets/ae187171-b99e-408e-81df-973bfc65ccb6

</details> 

cc @eh2077 @shawnborton

**Comment by rezkiy37** [comment]
I've replaced `withOnyx` with `useOnyx`.

cc @eh2077

**Comment by mountiny** [comment]
@eh2077 can you please retest? Thanks!

**Comment by eh2077** [comment]
> I've replaced `withOnyx` with `useOnyx`.
> 

@rezkiy37 Why do we need this change? Does it bring benefits for solving the issue here?

**Comment by eh2077** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [ ] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [ ] I verified that all the inputs inside a form are aligned with each other.
    - [ ] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by eh2077** [comment]
@rezkiy37 Mobile Safari backdrop attachment picker modal still focus the input

https://github.com/user-attachments/assets/3f9f9927-444e-467a-9089-731335bff639

**Comment by rezkiy37** [comment]
> > I've replaced `withOnyx` with `useOnyx`.
> 
> @rezkiy37 Why do we need this change? Does it bring benefits for solving the issue here?

https://callstack-hq.slack.com/archives/C01GTK53T8Q/p1726169039084589

Screenshot 2024-09-18 at 21 11 18

**Comment by rezkiy37** [comment]
Hello!
I am OOO from 19.09 - 23.09. I will continue to work on this one when I am back.

**Comment by JKobrynski** [comment]
@eh2077 Hi, I'm Julian from Callstack and I'm taking care of this PR in @rezkiy37 's absence. I tried reproducing that on my end and it seems to work correctly. Are you still experiencing it?

https://github.com/user-attachments/assets/7132fde7-51da-45fd-9397-d0c734793953

**Comment by eh2077** [comment]
@JKobrynski Thanks for checking this out! In your clip, I saw the composer border is highlighted. Is it related to the composer focusing?

I'll test on my physical iPhone and update soon.

**Comment by rezkiy37** [comment]
I am back so I will manage the PR 🙂

**Comment by rezkiy37** [comment]
Hey!
I've synced up the branch with the latest `main`. Fortunately, I cannot reproduce the bug:

> Select an emoji won't open the keyboard, this should be unexpected according to https://github.com/Expensify/App/pull/48415#issuecomment-2357907633

<details><summary>Details</summary>
<p>


https://github.com/user-attachments/assets/666b6a64-9850-491c-be17-b3481502b249


</p>
</details> 

However, this one is replicable: 

> Tested on physical mobile Safari, backdrop attachment picker modal will not pop up the keyboard, but the composer has a highlight border.

So I am working to fix and test it. 

cc @eh2077 @JKobrynski

**Comment by rezkiy37** [comment]
I've fixed when the backdrop attachment picker modal will not pop up the keyboard, but the composer has a highlight border. Commit - a14bd8f.

<details><summary>Video</summary>

https://github.com/user-attachments/assets/f871fa06-a3fb-46a2-8947-486a2d8d30ac

</details>

**Comment by mountiny** [comment]
@rezkiy37 let us know when this is ready for another full look by c+

**Comment by rezkiy37** [comment]
@mountiny, this is ready. All comments have been addressed 🙂

**Comment by eh2077** [comment]
Cool! I'll retest it by the EOD

**Review by eh2077 — APPROVED** [review]
LGTM and tested well! 

@rezkiy37 Great job and thank you for your patience!

**Comment by eh2077** [comment]
@mountiny All yours

**Review by mountiny — DISMISSED** [review]
Thanks for your patience @rezkiy37 and great testing @eh2077

**Review by mountiny — APPROVED** [review]

**Pull request merged**